### PR TITLE
Fix transforming tensor bug in bwd of batch_norm op and add more tests

### PR DESCRIPTION
**Pull request by Jie-Fang**
1. fix bug of transforming tensor in BatchNormGradKernel::Compute, add x dims constraint. BatchNormKernel::Compute has added, but forgot to add in GradKernel.

2. fix bug in batchnorm unittest, change data_format to data_layout.

3. add more situations on amp unittest in order to cover batchnorm op test.

### CODE REVIEW

**Comment by Jie-Fang** [comment]
4. move resize and transform function to transform.h to reduce the code reuse in the batchnorm and conv2d.

File: python/paddle/fluid/nets.py [hunk 1]
```diff
@@ -146,7 +146,8 @@ def img_conv_group(input,
                    conv_batchnorm_drop_rate=0.0,
                    pool_stride=1,
                    pool_type="max",
-                   use_cudnn=True):
+                   use_cudnn=True,
+                   data_format="NCHW"):
```
Comment (lanxianghit) [inline python/paddle/fluid/nets.py:150]: 新增参数data_format的默认值是NCHW，是否与之前的默认行为相符？

Comment (gongweibao) [reply python/paddle/fluid/nets.py:150]: 相符。
